From d5ef57f1efc967fbc88dfa0f0177be354caf039b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michel=20D=C3=A4nzer?= <mdaenzer@redhat.com>
Date: Tue, 26 Sep 2023 17:23:32 +0200
Subject: [PATCH xserver] test: Skip Xwayland test early if PIGLIT_DIR /
 XTEST_DIR isn't set

No point starting weston and waiting for it to start up in that case.
---
 test/scripts/xwayland-piglit.sh | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/test/scripts/xwayland-piglit.sh b/test/scripts/xwayland-piglit.sh
index 97a1759d9..9d6e54d01 100755
--- a/test/scripts/xwayland-piglit.sh
+++ b/test/scripts/xwayland-piglit.sh
@@ -1,5 +1,17 @@
 #!/bin/bash -e
 
+if test "x$XTEST_DIR" = "x"; then
+    echo "XTEST_DIR must be set to the directory of the xtest repository."
+    # Exit as a "skip" so make check works even without xtest.
+    exit 77
+fi
+
+if test "x$PIGLIT_DIR" = "x"; then
+    echo "PIGLIT_DIR must be set to the directory of the piglit repository."
+    # Exit as a "skip" so make check works even without piglit.
+    exit 77
+fi
+
 # this times out on Travis, because the tests take too long.
 if test "x$TRAVIS_BUILD_DIR" != "x"; then
     exit 77
-- 
2.43.2

